File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/lockable/LockableShowCodeDialogMessage.scala





















// Generated on 05/03/2013 10:34:45
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait LockableShowCodeDialogMessage extends MessageDeserializer {
  val messageId = 5740

  protected trait Message extends BaseMessage with Serializable {
    val messageId = LockableShowCodeDialogMessage.messageId

    val changeOrUse: Boolean
    val codeSize: Byte

    def serialize(buf: Buffer) {
      buf.writeBoolean(changeOrUse)
      buf.writeByte(codeSize)
    }

  }

  protected def create(changeOrUse: Boolean, codeSize: Byte): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val changeOrUse = buf.readBoolean()
    val codeSize = buf.readByte()

    create(changeOrUse, codeSize)
  }

}

object LockableShowCodeDialogMessage extends LockableShowCodeDialogMessage {
  case class MessageImpl(changeOrUse: Boolean, codeSize: Byte) extends Message
  type Target = MessageImpl

  override def create(changeOrUse: Boolean, codeSize: Byte) = MessageImpl(changeOrUse, codeSize)

  def apply(changeOrUse: Boolean, codeSize: Byte) = create(changeOrUse, codeSize)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/LockableShowCodeDialogMessage.java


// Generated on 05/08/2013 19:37:47
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class LockableShowCodeDialogMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5740;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public boolean changeOrUse;
    public byte codeSize;
    
    public LockableShowCodeDialogMessage() { }
    
    public LockableShowCodeDialogMessage(boolean changeOrUse, byte codeSize) {
        this.changeOrUse = changeOrUse;
        this.codeSize = codeSize;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeBoolean(changeOrUse);
        buf.writeByte(codeSize);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        changeOrUse = buf.readBoolean();
        codeSize = buf.readByte();
        if (codeSize < 0)
            throw new RuntimeException("Forbidden value on codeSize = " + codeSize + ", it doesn't respect the following condition : codeSize < 0");
    }
    
}
